subprojects/language-native/src/main/java/org/gradle/language/swift/SwiftVersion.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.language.swift;
+
+import org.gradle.api.Incubating;
+import org.gradle.util.VersionNumber;
+
+/**
+ * Swift language version.
+ *
+ * @since 4.5
+ */
+@Incubating
+public enum SwiftVersion {
+    SWIFT2, SWIFT3, UNKNOWN, SWIFT4;
```

Comment:
I wouldn't have 'unknown' as a possible Swift version value. It's actually the Swift _platform_ that is unknown.